2 Business layer/CandidateEvaluator.Core/Handlers/Queries/GetAllUserActivitiesHandler.cs

```diff
@@ -0,0 +1,28 @@
+﻿using CandidateEvaluator.Contract.Handlers;
+using CandidateEvaluator.Contract.Models;
+using CandidateEvaluator.Contract.Queries.UserActivity;
+using CandidateEvaluator.Contract.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CandidateEvaluator.Core.Handlers.Queries
+{
+    public class GetAllUserActivitiesHandler : IQueryHandler<GetAllUserActivities, List<RecentActivity>>
+    {
+        private readonly IUserRecentActivityService _activityService;
+
+        public GetAllUserActivitiesHandler(IUserRecentActivityService activityService)
+        {
+            _activityService = activityService;
```

Comment:
I believe that handlers should operate straight on repositories/context. The service layer is redundant here.